src/python/pyproject.rs

```diff
@@ -0,0 +1,69 @@
+use anyhow::{Context, Result};
+
+use crate::{chain, nixpacks::app::App};
+
+pub struct ProjectMeta {
```

Comment:
I think a lot of the chaining below can be cleaned up by deriving the Deserialize traits on this struct. Similar to the rust Cargo.toml format https://github.com/railwayapp/nixpacks/blob/main/src/providers/rust.rs#L13.

Reply:
Correct me if I'm wrong, but I think that in this case it might actually make more sense to do the chaining:
1. It only needs access to a few of the fields in the file, so deserializing the entire data structure might actually add extra overhead.
2. There are fallbacks for each field, which seems like the sort of thing that chaining (specifically with `.or_else`) is useful for.

Plus, it's not nearly as painful with a macro.